### PR TITLE
FEAT/ new face payment check and reset balance

### PR DESCRIPTION
**Pull request by rizkybaihaqy**
## Type of changes

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Enhancement
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behaviour?

we call api for checking phone number and then backend decide if the user required to fill out fill form.

## What is the new behavior?

- we move the checking for phone number and is pin required to client (previously on the server)

if balance insufficient now user can:
- reset balance
- back to catalog
- finish the demo

## Checklist

- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have filled the `[Unreleased]` [Changelog](https://github.com/nodefluxio/weber/blob/main/CHANGELOG.md

## Security Pentester API Checklist
- [ ] IDOR
- [ ] Malicious Script Injection
- [ ] SQL Injection

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: frontend/app/components/modules/PaymentPay/PaymentPay.tsx [hunk 1]
```diff
@@ -151,35 +182,68 @@ export const PaymentPay = ({ sessionId, amount, afterPay }: Props) => {
...
+                    )
+                  )
+                }}>
+                Confirm
+              </Button>
+            </>
+          ) : (
+            <>
+              <p className={styles.balance}>
+                <strong>Sorry, your balance is not enough</strong>
```
Comment (gregoriusjimmy) [inline frontend/app/components/modules/PaymentPay/PaymentPay.tsx:221]: @rizkybaihaqy task yg ini berarti udah kelar disini ya ki? 
image

**Comment by williammfu** [comment]
image

ky, display balancenya pake `formatMoney` nya `utils.ts` aja biar bagus hehe

File: frontend/app/components/modules/PaymentPay/PaymentPay.tsx (hunk 1 above)
Comment (rizkybaihaqy) [reply frontend/app/components/modules/PaymentPay/PaymentPay.tsx:221]: iyak

**Comment by rizkybaihaqy** [comment]
@williammfu okok

**Review by sonadztux — APPROVED** [review]

**Review by gregoriusjimmy — APPROVED** [review]
GOKIL @valdmir

**Pull request merged**